src/pages/docs/react-integration.mdx

```diff
@@ -37,21 +34,253 @@ The `JsonForms` component takes the following props:
+You can see a generated schema example in [our Examples section](../examples/gen-both-schemas).
+The generated schema is useful for rapid prototyping, but generally it is preferred to provide your own schema.
+
+An example schema would look like this:
+
+```
+const schema = {
+  type: 'object',
+  properties: {
+    name: {
+      type: 'string',
+      minLength: 1,
+    },
+    description: {
+      type: 'string',
+    },
+    done: {
+      type: 'boolean',
+    },
+    due_date: {
+      type: 'string',
+      format: 'date',
+    },
+    rating: {
+      type: 'integer',
+      maximum: 5,
+    },
+    recurrence: {
+      type: 'string',
+      enum: ['Never', 'Daily', 'Weekly', 'Monthly'],
+    },
+    recurrence_interval: {
+      type: 'integer',
+    },
+  },
+  required: ['name', 'due_date'],
+};
+```
+
+## `uischema`
+
+The `uischema` prop is a JSON describing the layout of the form. It can contain different UI schema elements, such as layouts
+and controls as well as rules for dynamically controlling different features of the UI elements based on data.
+You can find more information on the different UI elements and rules [here](/docs/uischema).
+
+If no UI schema is provided, JSON Forms will generate one for you. You can find an example
+of a generated UI schema in [our Examples section](../examples/gen-uischema).
+
+An example UI schema for the schema defined above would look like this:
+
+```
+{
+  "type": "VerticalLayout",
+  "elements": [
+    {
+      "type": "Control",
+      "label": false,
+      "scope": "#/properties/done"
+    },
+    {
+      "type": "Control",
+      "scope": "#/properties/name"
+    },
+    {
+      "type": "HorizontalLayout",
+      "elements": [
+        {
+          "type": "Control",
+          "scope": "#/properties/due_date"
+        },
+        {
+          "type": "Control",
+          "scope": "#/properties/rating"
+        }
+      ]
+    },
+    {
+      "type": "Control",
+      "scope": "#/properties/description",
+      "options": {
+        "multi": true
+      }
+    },
+    {
+      "type": "HorizontalLayout",
+      "elements": [
+        {
+          "type": "Control",
+          "scope": "#/properties/recurrence"
+        },
+        {
+          "type": "Control",
+          "scope": "#/properties/recurrence_interval",
+          "rule": {
+            "effect": "HIDE",
+            "condition": {
+              "type": "LEAF",
+              "scope": "#/properties/recurrence",
+              "expectedValue": "Never"
+            }
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+## `data`
+
+The `data` prop represents an object containing the data to be rendered in the form.
+
+<Note
+  title='ATTENTION'
+  body='Never update the data prop except when you intentionally want to set new data.'
+/>
+
+The data given to JSON Forms should only be updated when necessary, for example, when clearing a form:
```

Comment:
This is obsolete. The warning can be removed and all examples be updated so that they update the data. Actually this is now the more stable approach so we should encourage it.